### PR TITLE
fix: Fix border styling for buttons in Calendar and fix arrow navigation in CalendarDayGrid and CalendarMonthPicker

### LINKED ISSUE #26237: DatePicker: Update getFocusStyles calls to work in v9
Issue to track the conversion of getFocusStyles to v9 styles.

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

Buttons in the Calendar have a black border and using arrow navigation in the Calendar's day grid and Calendar's Month picker is not working.

## New Behavior

Buttons in the Calendar no longer have a border and arrow navigation now works correctly for both Calendar's day grid and Calendar's month picker

## Related Issue(s)

- Fixes #26237

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/020829cf01db3ca0baffcd0dc640a34a91b548d3'>020829cf01db3ca0baffcd0dc640a34a91b548d3</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarDayGrid.tsx [hunk 1]
```diff
@@ -127,53 +127,49 @@ export const CalendarDayGrid: React.FunctionComponent<CalendarDayGridProps> = pr
...
   } as const;
 
-  const arrowNavigationAttributes = useArrowNavigationGroup({
-    axis: 'both',
-    ignoreDefaultKeydown: { Enter: true, Escape: true, Tab: true },
-  });
+  const arrowNavigationAttributes = useArrowNavigationGroup({ axis: 'grid' });
 
   return (
-    <div {...arrowNavigationAttributes}>
```
Comment (sopranopillow) [inline packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarDayGrid.tsx:136]: This is a minor change but gh takes it as bigger than it is 😅 just removing the wrapping div and adding the arrow navigation directly to the table.

File: packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarGridDayCell.tsx [hunk 2]
```diff
@@ -85,10 +85,10 @@ export const CalendarGridDayCell: React.FunctionComponent<CalendarGridDayCellPro
       direction = -1;
     } else if (ev.key === ArrowDown) {
       targetDate = addWeeks(date, 1);
-    } else if (getRTLSafeKey(ArrowLeft, dir)) {
+    } else if (ev.key === getRTLSafeKey(ArrowLeft, dir)) {
```
Comment (sopranopillow) [inline packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarGridDayCell.tsx:88]: This is the cause of the month change when tabbing out of the day grid 🙃

File: packages/react-components/react-datepicker/src/components/CalendarMonth/CalendarMonth.tsx [hunk 3]
```diff
@@ -246,51 +245,49 @@ export const CalendarMonth: React.FunctionComponent<CalendarMonthProps> = props
           </button>
         </div>
       </div>
-      <div {...arrowNavigationAttributes}>
```
Comment (sopranopillow) [inline packages/react-components/react-datepicker/src/components/CalendarMonth/CalendarMonth.tsx:249]: same as above, just moving the arrow navigation to the child and removing wrapping div

File: packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarGridDayCell.tsx (hunk 2 above)
Comment (khmakoto) [reply packages/react-components/react-datepicker/src/components/CalendarDayGrid/CalendarGridDayCell.tsx:88]: Ahhh, I'm very sorry for that, probably my fault...

**Review by khmakoto — APPROVED** [review]

**Pull request merged**